hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -317,10 +317,20 @@ def __repr__(self):
         bits = []
         for name, setting in all_settings.items():
             value = getattr(self, name)
+            # The only settings that are not shown are those that are
+            # deprecated and left at their default values.
             if value != setting.default or not setting.hide_repr:
                 bits.append('%s=%r' % (name, value))
         return 'settings(%s)' % ', '.join(sorted(bits))
 
+    def show_changed(self):
+        bits = []
+        for name, setting in all_settings.items():
+            value = getattr(self, name)
+            if value != setting.default or not setting.hide_repr:
```

Comment:
For this use (unlike the repr!) we only want the changed settings, so `if value != setting.default:` is fine without any mention of hidden settings.

Reply:
Yeah. I just based this on pytest-benchmark which displays all settings. Happy to change. I might make a separate issue to bump up what settings are displayed based on the verbosity if that sounds good to you?